portfolio/src/main/webapp/script.js

```diff
@@ -103,3 +103,28 @@ function createListElement(text) {
   liElement.innerText = text;
   return liElement;
 }
+
+/**
+ * Fetch the login status and unhide the comment form if the user is logged in
+ * Otherwise, show a login link
+ */
+function updateCommentForm() {
+  fetch('/login-status').then(response => response.text()).then(result => {
+    result = result.slice(3, result.length - 5);
+    if (result[0] === '1') {
+      // the user is logged in, then unhide commentForm and show logout url
+      document.getElementById('commentForm').style.visibility = 'visible';
+      displayLink("logoutLink");
+    } else {
+      // show login url
+      displayLink("loginLink");
+    }
+  });
+}
```

Comment:
So this is checking the login status API, updating the comment form, *and* it is updating the login/logout links even though it is called `updateCommentForm` :-)  

It might make sense to split this up a bit so that you have smaller functions that are responsible for just one thing.  This makes it easier to maintain, change, re-use, and understand the code both for people you are working with and even yourself (e.g. imagine if you came back to this in 6 months time - you might have forgotten all about it). 

For example you can have individual functions for showing/hiding the various bits - you were already on the right track with `displayLink` so that was a great start:

```
function displayCommentForm(isShown) {
  // display the comment form if isShown === true, otherwise hide it
}

function displayLoginLink(isShown) {
  // display the login link if isShown === true, otherwise hide it
}

function displayLogoutLink(isShown) {
  // display the logout link if isShown === true, otherwise hide it
}
```

And then have a single function that is responsible for checking the login status that calls these smaller functions, e.g.

```
function updateVisibilityForLoginStatus() {
  // Default to hiding everything first
  displayLoginLink(false);
  displayLogoutLink(false);
  displayCommentForm(false);

  // Get the login status for the user and only show the appropriate bits
  fetch('/login-status').then(response => response.text()).then(result => {
    if (result[0] === '1') {
      displayLogoutLink(true);
      displayCommentForm(true);
    } else {
      displayLoginLink(true);
    }
  }
}
```

Reply:
I changed the function name, but is it ok to use `displayElement(elementId, isShown) `instead of `displayLoginLink(isShown)` and `displayLogoutLink(isShown)`?

Reply:
Yep sure of course :)  These are just my views and opinions - a review is always a two-way conversation.